### PR TITLE
Feature/vta 341

### PR DESCRIPTION
**Pull request by wis-dvsa**
regression pack fix work

### CODE REVIEW

File: src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java [hunk 1]
```diff
@@ -382,6 +386,7 @@ public void testResults_Roadworthiness_TRL_QKT_Pass_Certificate_Generation() {
         testResultsSteps.postVehicleTestResultsWithAlterations(testResultRecord, alterations);
         testResultsSteps.statusCodeShouldBe(201);
         testResultsSteps.validateData("Test records created");
+        ;
```
Comment (blondeholly) [inline src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java:389]: Can this be removed please

File: src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java [hunk 2]
```diff
@@ -845,6 +853,7 @@ public void testResults_Roadworthiness_HGV_QJV5_Fail_Certificate_Generation() {
         testResultsSteps.validateCertificateIsGenerated(testNumber,randomVin);
     }
 
+
```
Comment (blondeholly) [inline src/test/java/testresults/PostTestResultsRoadworthinessCertificateGeneration.java:856]: Can this line be removed please

File: src/test/java/testresults/PutTestResultsCertificateGeneration.java [hunk 3]
```diff
@@ -10,7 +10,8 @@
 import net.thucydides.core.annotations.Title;
 import net.thucydides.core.annotations.WithTag;
 import net.thucydides.junit.annotations.TestData;
-import org.apache.commons.lang3.time.DateUtils;
+import org.apache.commons.lang3.time.DateUtils;;
```
Comment (blondeholly) [inline src/test/java/testresults/PutTestResultsCertificateGeneration.java:13]: can the extra semi colon be removed please

File: src/test/java/testresults/PutTestResultsCertificateGeneration.java [hunk 4]
```diff
@@ -19,22 +20,23 @@
 import util.JsonPathAlteration;
 import org.apache.http.HttpStatus;
 
-import java.text.SimpleDateFormat;
+import java.text.SimpleDateFormat;;
```
Comment (blondeholly) [inline src/test/java/testresults/PutTestResultsCertificateGeneration.java:23]: can the extra semi colon be removed please

File: src/test/java/trailerRegistration/PostTrailerRegistration.java [hunk 5]
```diff
@@ -5,6 +5,7 @@
 import net.thucydides.core.annotations.Steps;
 import model.vehicles.VehicleTechnicalRecordStatus;
 import net.thucydides.core.annotations.Title;
+import net.thucydides.core.annotations.WithTag;
```
Comment (blondeholly) [inline src/test/java/trailerRegistration/PostTrailerRegistration.java:8]: please remove unused with tag

**Review by blondeholly — CHANGES_REQUESTED** [review]
Hi Wisdom, We need to speak to Iulia about the json dates, should they all be 2020 or should some be 2021 and what is the rule. Some of these may be ni picky, I apologise. Dan has told me off. :)

File: src/main/java/steps/VehicleTechnicalRecordsSteps.java [hunk 6]
```diff
@@ -613,4 +616,62 @@ public void validateMessage(String stringData) {
...
+                noVehicles = response.then().extract().jsonPath().getInt("$.size()");
+            }
+            catch(Exception e) {
+                System.out.println( e + " " + "NO TECH RECORD IS RETURNED");
+            }
+            for (int j = 0; j < noVehicles; j++) {
+
+                int recordsNumber = response.then().log().all().extract().jsonPath().get("[" + j + "].techRecord.size()");
+                for (int k = 0; k < recordsNumber; k++) {
+//                    String createdAtString = response.then().log().all().extract().jsonPath().get("[" + j + "].techRecord[" + k + "].createdAt");
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/java/steps/VehicleTechnicalRecordsSteps.java]: Done

File: src/main/resources/loader/develop/technical-records_post_payload_lgv_10328.json [hunk 7]
```diff
@@ -6,7 +6,7 @@
   "primaryVrm": "QU123RT",
   "techRecord": [
     {
-      "createdAt": "2019-06-24T10:26:54.903Z",
+      "createdAt": "2021-06-24T10:26:54.903Z",
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/resources/loader/develop/technical-records_post_payload_lgv_10328.json:9]: ignoring as per chat

File: src/main/resources/loader/develop/technical-records_post_payload_motorcycle_10328.json [hunk 8]
```diff
@@ -6,7 +6,7 @@
   "primaryVrm": "QU123RT",
   "techRecord": [
     {
-      "createdAt": "2019-06-24T10:26:54.903Z",
+      "createdAt": "2021-06-24T10:26:54.903Z",
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/resources/loader/develop/technical-records_post_payload_motorcycle_10328.json:9]: ignoring as per chat

File: src/main/resources/loader/develop/technical-records_put_payload_lgv_11814.json [hunk 9]
```diff
@@ -7,7 +7,7 @@
   "primaryVrm": "73XIZ00",
   "techRecord": [
     {
-      "createdAt": "2019-06-24T10:26:54.903Z",
+      "createdAt": "2021-06-24T10:26:54.903Z",
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/resources/loader/develop/technical-records_put_payload_lgv_11814.json:10]: ignoring as per chat

File: src/main/resources/loader/develop/technical-records_put_payload_motorcycle_11814.json [hunk 10]
```diff
@@ -7,7 +7,7 @@
   "primaryVrm": "73XIZ00",
   "techRecord": [
     {
-      "createdAt": "2019-06-24T10:26:54.903Z",
+      "createdAt": "2021-06-24T10:26:54.903Z",
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/resources/loader/develop/technical-records_put_payload_motorcycle_11814.json:10]: ignoring as per chat

File: src/test/java/testresults/PutTestResultsCertificateGeneration.java (hunk 3 above)
Comment (wis-dvsa) [reply src/test/java/testresults/PutTestResultsCertificateGeneration.java]: Done

File: src/test/java/testresults/PutTestResultsCertificateGeneration.java (hunk 4 above)
Comment (wis-dvsa) [reply src/test/java/testresults/PutTestResultsCertificateGeneration.java]: Done

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java [hunk 11]
```diff
@@ -268,48 +268,52 @@ public TestPostTestResultsAnnualCertificateGenerationTrl(String name, String tes
     @Title("CVSB-8798 - Annual certificate is generate for all Trl tests")
     @Test
     public void testResults_Annual_Certificate_Generation_Trl() {
+            // Read the base test result JSON.
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java]: Done

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java [hunk 12]
```diff
@@ -268,48 +268,52 @@ public TestPostTestResultsAnnualCertificateGenerationTrl(String name, String tes
...
-        testResultsSteps.validateData("Test records created");
-        testResultsSteps.getTestResults(randomSystemNumber);
-        testResultsSteps.statusCodeShouldBe(200);
-        String testNumber = testResultsSteps.getTestNumber();
-        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", testCode);
-        Assert.assertTrue(testResultsSteps.validateCertificateNumberLength());
+            // Post the results, together with any alterations, and verify that they are accepted.
+            testResultsSteps.postVehicleTestResultsWithAlterations(testResultRecord, alterations);
+            if ("warmup test".equals(name)) {
+                testResultsSteps.statusCodeShouldBe(504);
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java]: Done

File: src/test/java/trailerRegistration/PostTrailerRegistration.java (hunk 5 above)
Comment (wis-dvsa) [reply src/test/java/trailerRegistration/PostTrailerRegistration.java]: Done

File: src/main/resources/loader/develop/technical-records_trl_all_fields.json [hunk 13]
```diff
@@ -14,8 +14,8 @@
   "techRecord": [
     {
       "vehicleType": "trl",
-      "regnDate": "2019-06-25",
-      "firstUseDate": "2019-06-25",
+      "regnDate": "2021-06-25",
+      "firstUseDate": "2021-06-25",
```
[the remark this answers is not in the document]
Comment (IuliaDVSA) [reply src/main/resources/loader/develop/technical-records_trl_all_fields.json:18]: Please see above

File: src/main/resources/loader/develop/technical-records_trl_all_fields.json [hunk 14]
```diff
@@ -36,15 +36,15 @@
       "alterationMarker": false,
       "departmentalVehicleMarker": true,
       "authIntoService": {
-        "cocIssueDate": "2019-06-25",
-        "dateReceived": "2019-06-25",
-        "datePending": "2019-06-25",
-        "dateAuthorised": "2019-06-25",
-        "dateRejected": "2019-06-25"
+        "cocIssueDate": "2021-06-25",
```
[the remark this answers is not in the document]
Comment (IuliaDVSA) [reply src/main/resources/loader/develop/technical-records_trl_all_fields.json:39]: Please see above

**Review by IuliaDVSA — CHANGES_REQUESTED** [review]

File: src/main/java/util/WebDriverBrowsertack.java [hunk 15]
```diff
@@ -95,6 +95,7 @@ public static WebDriver checkVsaEmail(String randomVrm) {
                 ExpectedConditions.presenceOfElementLocated(By.cssSelector("input[name='loginfmt']"))
         ));
         driver.findElement(By.cssSelector("input[name='loginfmt']")).sendKeys(loader.getMicrosoftOnlineUserName());
+
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/java/util/WebDriverBrowsertack.java:98]: this does require any formatting

[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/java/util/WebDriverBrowsertack.java:98]: removed blank line

File: src/main/java/util/WebDriverBrowsertack.java [hunk 16]
```diff
@@ -104,6 +105,7 @@ public static WebDriver checkVsaEmail(String randomVrm) {
                     ExpectedConditions.presenceOfAllElementsLocatedBy(By.cssSelector("input[name='passwd']"))
             ));
             driver.findElement(By.cssSelector("input[name='passwd']")).sendKeys(loader.getMicrosoftOnlinePass());
+
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/java/util/WebDriverBrowsertack.java:108]: removed blank line

File: src/main/java/steps/VehicleTechnicalRecordsSteps.java [hunk 17]
```diff
@@ -18,6 +18,9 @@
 import util.AwsUtil;
 import static util.TypeLoader.*;
 import util.JsonPathAlteration;
+
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/main/java/steps/VehicleTechnicalRecordsSteps.java:21]: Done

File: src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java [hunk 18]
```diff
@@ -268,48 +268,52 @@ public TestPostTestResultsAnnualCertificateGenerationTrl(String name, String tes
...
 
-        JsonPathAlteration alterationVin = new JsonPathAlteration("$.vin", randomVin, "", "REPLACE");
-        JsonPathAlteration alterationSystemNumber = new JsonPathAlteration("$.systemNumber", randomSystemNumber, "", "REPLACE");
-        JsonPathAlteration alterationTestResultId = new JsonPathAlteration("$.testResultId", randomTestResultId, "", "REPLACE");
-        JsonPathAlteration alterationTestName = new JsonPathAlteration("$.testTypes[0].name", name,"","REPLACE");
-        JsonPathAlteration alterationTestTypeId = new JsonPathAlteration("$.testTypes[0].testTypeId", testTypeId,"","REPLACE");
-        JsonPathAlteration alterationTestTypeName = new JsonPathAlteration("$.testTypes[0].testTypeName", testTypeName,"","REPLACE");
-        JsonPathAlteration alterationTestResult = new JsonPathAlteration("$.testTypes[0].testResult", testResult,"","REPLACE");
-        JsonPathAlteration alterationNoOfAxles = new JsonPathAlteration("$.noOfAxles", noOfAxles,"","REPLACE");
+            JsonPathAlteration alterationVin = new JsonPathAlteration("$.vin", randomVin, "", "REPLACE");
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/test/java/testresults/annual_certificates/TestPostTestResultsAnnualCertificateGenerationTrl.java]: done

File: src/test/java/vehicle/PostVehicleTechnicalRecords.java [hunk 19]
```diff
@@ -364,6 +375,7 @@ public void testCreatePsvWitSpecialCharacters_VIN() {
         // TEST
         vehicleTechnicalRecordsSteps.postVehicleTechnicalRecordsWithAlterations(postRequestBody, alterations);
         vehicleTechnicalRecordsSteps.statusCodeShouldBe(201);
+        System.out.println("RANDOM" + "" + randomVin);
```
[the remark this answers is not in the document]
Comment (wis-dvsa) [reply src/test/java/vehicle/PostVehicleTechnicalRecords.java:378]: done

**Review by IuliaDVSA — APPROVED** [review]

**Review by blondeholly — DISMISSED** [review]

**Review by blondeholly — APPROVED** [review]

**Review by danielchildsbjss — APPROVED** [review]
Nice work

**Review by blondeholly — APPROVED** [review]

**Pull request merged**